### ISSUE TITLE
Use ijson to remove requirement for newline-delimited JSON

### ISSUE THREAD

**Issue by jpmckinney** [open-contracting/ocdskit#104]
Requiring newline-delimited JSON is an extra hurdle. We can do one or both of:

1. Use an iterative JSON parser like `ijson` to read arrays instead of requiring a stream as input
1. Find a [concatenated JSON](https://en.wikipedia.org/wiki/JSON_streaming#Concatenated_JSON) parser to read streams even if they are not newline-delimited

We can also use `ijson.items()` to replace use of `jq`, which is mostly used to generate a stream from deep arrays.

**Comment by jpmckinney** [comment]
ijson can parse concatenated JSON: 

```python
from io import BytesIO
import ijson.backends.yajl2_cffi as ijson

stream = BytesIO(b'{"foo":"bar"}{"baz":"bzz"}')
for item in ijson.common.items(ijson.parse(stream, multiple_values=True), ''):
    print(repr(item))

# {'foo': 'bar'}
# {'baz': 'bzz'}
```

From https://github.com/isagalaev/ijson/issues/75

**Comment by jpmckinney** [comment]
FYI, the `json` module is faster, so if we need to keep a fast option, we can add a flag for people to opt-in to line-delimited JSON.

**Comment by duncandewhurst** [comment]
![](https://github.trello.services/images/mini-trello-icon.png) [ocdskit compile: improve support for large files](https://trello.com/c/ldVF0B3y/94-ocdskit-compile-improve-support-for-large-files)

**Comment by rtobar** [comment]
Please beware that with the latest ijson version there is a `yajl2_c` backend that is 10x faster that `yajl2_cffi`

**Comment by jpmckinney** [comment]
Thanks @rtobar! We currently can't use yajl2_c due to the Decimal bug (which is fixed, but not released in a new version). When do you expect a new version to be available?

**Comment by jpmckinney** [comment]
Closed by d33790e743333233da18b57c607817d9011936aa

**Comment by rtobar** [comment]
@jpmckinney Good point. I could be putting up a new release next week probably, there are already a couple of changes worth seeing the light of day (including better default backend selection).
